contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,944 @@
+	}
+
+	/**
+	 * @dev Borrow base asset and transfer to 'to' account.
+	 * @notice Borrow base token from Compound on behalf of an address.
+	 * @param market The address of the market.
+	 * @param to The address to which the borrowed asset is transferred.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowOnBehalf(
+		address market,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		(uint256 amt_, uint256 setId_) = _borrowOrWithdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: getBaseToken(market),
+				from: address(0),
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+		eventName_ = "LogBorrowOnBehalf(address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, to, amt_, getId, setId_);
+	}
+
+	/**
+	 * @dev Borrow base asset from 'from' and transfer to 'to'.
+	 * @notice Borrow base token or deposited token from Compound.
+	 * @param market The address of the market.
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param from The address from where asset is to be withdrawed.
+	 * @param to The address to which the borrowed assets are to be transferred.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowFrom(
+		address market,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		(uint256 amt_, uint256 setId_) = _borrowOrWithdraw(
+			BorrowWithdrawParams({
+				market: market,
+				token: getBaseToken(market),
+				from: from,
+				to: to,
+				amt: amt,
+				getId: getId,
+				setId: setId
+			})
+		);
+		eventName_ = "LogBorrowFrom(address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, from, to, amt_, getId, setId_);
+	}
+
+	/**
+	 * @dev Repays the borrowed base asset.
+	 * @notice Repays the borrow of the base asset.
+	 * @param market The address of the market.
+	 * @param amt The amount to be repaid.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function payback(
+		address market,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+		require(market != address(0), "invalid market address");
+
+		address token = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		amt_ = amt_ == uint256(-1)
+			? CometInterface(market).borrowBalanceOf(address(this))
```

Comment:
As market itself is a token, we can use this instead
```suggestion
			? TokenInterface(market).balanceOf(address(this))
```

Reply:
done